### PR TITLE
PORT-7618 | Dynatrace Ocean Integration with Invalid Entity Fields

### PR DESCRIPTION
**Pull request by lordsarcastic**
# Description

What - Fixed bug affecting Entity blueprints where fields: firstSeenTms, lastSeenTms, and tags; are missing
Why - So bugs won't be in the integration
How - Added query parameters to explicitly fetch these missing fields

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots
Screenshot 2024-04-17 at 18 40 16

## API Documentation
[Monitored Entity API](https://docs.dynatrace.com/docs/dynatrace-api/environment-api/entity-v2/get-entities-list#parameters)

### CODE REVIEW

**Review by PeyGis — CHANGES_REQUESTED** [review]
@lordsarcastic for every new updates to the integration, we need to bump the integration version as well as creating changelog for it. This is missing from this PR. Please include them

**Review by yairsimantov20 — APPROVED** [review]
🌊

File: integrations/dynatrace/client.py [hunk 1]
```diff
@@ -63,10 +63,15 @@ async def get_slos(self) -> AsyncGenerator[list[dict[str, Any]], None]:
     async def _get_entities_from_type(
         self, type_: str
     ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        ENTITY_FIELDS = "firstSeenTms,lastSeenTms,tags"
```
Comment (yairsimantov20) [inline integrations/dynatrace/client.py:66]: why not enabling it as part of the resource selector?

Comment (lordsarcastic) [reply integrations/dynatrace/client.py:66]: They are fields defined in the blueprint itself. If they are put in the resource selector, there is the possibility of users overriding it. By putting it here, we ensure they are included automatically. They are not meant to be customized.

Comment (yairsimantov20) [reply integrations/dynatrace/client.py:66]: Wdym? The blueprint is only a default option for modeling the data
The customer can and will change the blueprint to his needs
Therefore we need to allow him to pick which data he wants

Comment (lordsarcastic) [reply integrations/dynatrace/client.py:66]: If that's the case, which do you suggest:
- Take this out and make it as a default option on the selector
- Take this out, take the corresponding fields out in the default blueprint and add it to the selector.
- Any other option...

Comment (yairsimantov20) [reply integrations/dynatrace/client.py:66]: First option should be good

File: integrations/opencost/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+# Port_Ocean 0.1.9 (2024-04-26)
```
Comment (yairsimantov20) [inline integrations/opencost/CHANGELOG.md:10]: 0.1.9?

Comment (yairsimantov20) [reply integrations/opencost/CHANGELOG.md:10]: why is in the opencost?

File: integrations/dynatrace/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+# Port_Ocean 0.1.14 (2024-05-24)
```
Comment (yairsimantov20) [inline integrations/dynatrace/CHANGELOG.md:10]: update changelog date

File: integrations/dynatrace/client.py [hunk 4]
```diff
@@ -61,20 +61,26 @@ async def get_slos(self) -> AsyncGenerator[list[dict[str, Any]], None]:
...
+        self, type_: str, entity_fields: EntityFieldsType | None
     ) -> AsyncGenerator[list[dict[str, Any]], None]:
         async for entities in self._get_paginated_resources(
             f"{self.host_url}/entities",
             "entities",
-            params={"entitySelector": f'type("{type_}")', "pageSize": 100},
+            params={
+                "entitySelector": f'type("{type_}")',
+                "pageSize": 100,
+                "fields": entity_fields or "",
```
Comment (yairsimantov20) [inline integrations/dynatrace/client.py:72]: add the fields property only when u have entity fields

**Review by yairsimantov20 — CHANGES_REQUESTED** [review]

**Comment by lordsarcastic** [comment]
PR moved to [non fork](https://github.com/port-labs/ocean/pull/742)

**Pull request closed without merging**